app/assets/stylesheets/components/dough_theme/button/_button.scss

```diff
@@ -130,6 +130,7 @@
 // Styleguide Newsletter signup buttons
 
 .button--newsletter {
+  @extend .button;
```

Comment:
You probably meant to extend . newsletter-button from .button or rename .button-newsletter to .newsletter-button ?